fs_storage/models/fs_storage.py

```diff
@@ -266,12 +266,33 @@ def _compute_options_properties(self) -> None:
             doc = inspect.getdoc(cls.__init__)
             rec.options_properties = f"__init__{signature}\n{doc}"
 
+    def _get_marker_file_name(self):
+        return ".odoo_fs_storage_%s.marker" % self.id
+
+    def _check_connection(self, fs):
+        marker_file_name = self._get_marker_file_name()
+        try:
+            marker_file = fs.ls(marker_file_name, detail=False)
+            if not marker_file:
+                fs.touch(marker_file_name)
+        except FileNotFoundError:
+            fs.touch(marker_file_name)
+        return True
+
     @property
     def fs(self) -> fsspec.AbstractFileSystem:
         """Get the fsspec filesystem for this backend."""
         self.ensure_one()
         if not self.__fs:
             self.__fs = self._get_filesystem()
+        if not tools.config["test_enable"]:
+            # Check whether we need to invalidate FS cache or not.
+            # Use a marker file to limit the scope of the LS command for performance.
+            try:
+                self._check_connection(self.__fs)
+            except Exception as e:
```

Comment:
After some tests, we are still facing connections issues and the current implementation doesn't cover it: ```OSError: Socket is closed```
This errors arrives only after some time (several hours, workers have a high lifespan here and FS object is never recreated meanwhile).

Reply:
@sebalix maybe then we should not store the FS object at all in the slot, and always call to _get_filesystem

Reply:
The module can also be used in contexts of heavy traffic between odoo and external filesystems. This is particularly the case when it is used to manage Odoo's default attachment storage. In such circumstances, the cost of the reconnection required each time the fs proxy is used is not negligible. Perhaps it would be a good idea to set up a 'keep-alive' mechanism via a cron running every 'X' minutes...

Reply:
@lmignon we could do that or putting it as a configuration in the FS storage to decide if you want to check the connection everytime or not. The cron approach however seems to be the solution covering both cases, and one can decide to run it more/less frequently. What do you think @sebalix ?

Reply:
I understand it would be to have a cron that invalidates the __fs slot after some time

Reply:
The cron process could updates memory objects of other Odoo process (workers)? Not sure it's possible :thinking:

Reply:
The `raise e` is an issue here (testing the code during a workshop), it is blocking somehow the queue-job.
Clearing the fsspec cache should be enough here, so a new connection will be established in case of error.

Reply:
Ok, after one day of testing, we encountered no connection issues with the help of the `skip_instance_cache` parameter of `fsspec` (set in the options). This will add some overhead but at least we are safe... dealing with this cache in Odoo seems error prone (connection stability, configuration of the SFTP server...).

I don't know how huge is this overhead, load testing will tell us I guess.